src/EditorFeatures/Core/Shared/Utilities/IThreadingContext.cs

```diff
@@ -43,5 +46,28 @@ JoinableTaskFactory JoinableTaskFactory
         {
             get;
         }
+
+        /// <summary>
+        /// Gets a <see cref="CancellationToken"/> indicating that disposal has been requested for the threading
+        /// context.
+        /// </summary>
+        CancellationToken DisposalToken
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Runs an asynchronous operation. If the operation is not complete prior to the time the threading context is
+        /// disposed, the operation will block the shutdown.
```

Comment:
This strikes me as something of a substitute for a `JTF.RunAsync` call. I generally discourage facades over the JTF class itself since analyzers and the options available on extension method overloads can no longer add value to the caller.
Is this API really so much better than simply exposing `ShutdownBlockingTaskFactory` that is associated with the shutdown-blocking `JoinableTaskCollection` so callers can use the primary API? They already have access to the cancellation token as you're adding that to the interface, right?
For example, suppose you wanted to queue a shutdown-blocking operation with a particular main thread priority? You can do this if you have the JTF instance, but you can't with this method hiding it.

Reply:
I agree with this overall, but I'm leaning towards this form for the following:

* There is no "obviously superior" approach - meaning it functionally is the same or better and uses an API which, in the context of Roslyn, naturally leads to correct use of the API even for users who aren't familiar with JTF or shutdown sequences
* The functionality is not used in an especially large number of cases. Right now there are 3, and I believe there are 2-3 more that we want, but that's still not exceptionally common for this code base
* The functionality is currently captured by an API which would be easy to review and change in the future if the first point changes